matscipy/dislocation.py

```diff
@@ -2499,141 +2705,163 @@ def plot_unit_cell(self, ms=250, ax=None):
-
-        bulk = bulk[final_mask]
-
-        # disloc is a copy of bulk with displacements applied
-        disloc = bulk.copy()
-
-        disloc.positions += self.displacements(bulk.positions, shifted_center,
-                                               self_consistent=self_consistent)
-
-        r = np.sqrt(((bulk.positions[:, [0, 1]]
-                      - center[[0, 1]])**2).sum(axis=1))
-
-        fix_mask = r > radius - fix_width
-
-        shifted_r = np.sqrt(((bulk.positions[:, [0, 1]] -
-                              shifted_center[[0, 1]]) ** 2).sum(axis=1))
-
-        shifted_fix_max = shifted_r > radius - fix_width
-        extension = np.array(extension)
-        extended_center = center + extension
-        extended_r = np.sqrt(((bulk.positions[:, [0, 1]] -
-                               extended_center[[0, 1]]) ** 2).sum(axis=1))
-        extended_fix_max = extended_r > radius - fix_width
-        final_fix_mask = fix_mask & shifted_fix_max & extended_fix_max
+                       extension=np.array([0, 0, 0]),
+                       fix_width=10.0, self_consistent=None,
+                       method="atomman",
+                       ext_idxs = -1):
+        '''
+        Build dislocation cylinder for single dislocation system
 
-        x, y, _ = bulk.positions.T
-        x_mask = x - center[0] < extension[0] + core_position[0]
-        x_mask = x_mask * (x - center[0] > 0)
-        y_mask = np.abs(y - center[1]) > radius - fix_width
+        Parameters
+        ----------
+        radius: float
+            Radius of bulk surrounding the dislocation core
+        core_position: np.array
+            Vector offset of the core position from default site
+        extension: np.array
+            Add extra bulk to the system, to also surround a fictitious core
+            that is extension away from the center of the cell
+        fix_width: float
+            Defines a region to apply the FixAtoms ase constraint to
+            Fixed region is given by (radius - fix_width) <= r <= radius,
+            where the position r is measured from the dislocation core
+            (or from the glide path, if extra_glide_widths is given)
+        self_consistent: bool
+            Controls whether the displacement field used to construct the dislocation is converged 
+            self-consistently. If None (default), the value of self.self_consistent is used
+        use_atomman: bool
+            Use the Stroh solver included in atomman (requires atomman package) to
+            solve for displacements, or use the AnisotropicDislocation class
+        ext_idxs: int or np.array of ints
+            idxs of the cores each part of extension should apply to
+            extension = np.array([
+                            [-5.0, 0, 0],
+                            [5.0, 0, 0.0]
+                            ])
+            and ext_idxs = np.array([0, -1]) will extend the cell by 5Ang left of the first core, 
```

Comment:
Let's remove `ext_idxs` and replace with `np.arange(extension.shape[0])` - not much loss of functionality

Reply:
Removed ext_idxs as an arg - Instead, extension gives vector extensions from the first N cores. Very similar functionality, and probably much easier to understand.

I think i'd done the ext_idxs in case the user wanted to define two extensions from the same core (e.g. a dissociated dislocation where the partials both glide away), but in this case the user can hack into core_positions to make a custom script to do that themselves